scipy/spatial/_geometric_slerp.py

```diff
@@ -0,0 +1,206 @@
+from __future__ import division, print_function, absolute_import
+
+__all__ = ['geometric_slerp']
+
+import math
+import numpy as np
+from . import _slerp
+from scipy.spatial.distance import euclidean
+
+def geometric_slerp(start,
+                    end,
+                    t,
+                    tol=1e-7):
+    """
+    Spherical linear interpolation along unit-radius
+    great circle arc in arbitrary dimensional space.
+
+    Parameters
+    ----------
+    start : (D, ) array-like
+        Single D-dimensional input coordinate in a 1-D array-like
+        object.
+        Starting coordinate for interpolation will be converted
+        to float64.
+    end : (D, ) array-like
+        Single D-dimensional input coordinate in a 1-D array-like
+        object.
+        End coordinate for interpolation will be converted to
+        float64.
+    t: array
```

Comment:
Might add the shape description as `float or (n_points,) array-like`.

And `(D, )` I would maybe change to `(n_dimensions,)`